# AFRICAN STARTUPS' FUNDING

We are going to use an African startup dataset in an attempt to answer question(s) about African Startups.

**Importing important libraries for numeric manipulation and data viz**

In [2]:
#import all the important libraries
import pandas as pd
import numpy as  np
import matplotlib.pyplot as plt

**Loading the dataset, it is a .xlsx file**

In [3]:
#file location 
african_startups = r"C:\Users\user\Downloads\AfricanStartups\2021\Funded African tech startups 2021.xlsx"
#read and assign the xlsx to a df
df = pd.read_excel(african_startups)

In [5]:
#view df
df.head()

Startup  Unnamed: 1 Location         Sector                Date  \
0  Daystar Power         NaN  Nigeria         energy      January + July   
1        uLesson         NaN  Nigeria        ed-tech  January + December   
2      Cowrywise         NaN  Nigeria        fintech             January   
3         Ogelle         NaN  Nigeria  entertainment             January   
4         Contro         NaN  Nigeria       e-health             January   

   Amount (US$)                                          Investors  \
0    62000000.0  Investment Fund for Developing Countries (IFU)...   
1    22500000.0  Owl Ventures, TLcom Capital, Founder Collectiv...   
2     3000000.0  Quona Capital, Sahil Lavingia, Tsadik Foundati...   
3           NaN                            African Discovery Group   
4       42000.0                                                NaN   

                 Stage  Debt/Equity Female co-founder/CEO? Local founder?  \
0             Series B  Equity/debt                     No            Yes   
1  Series A / Series B       Equity                     No            Yes   
2         Pre-Series A       Equity                     No            Yes   
3          Undisclosed       Equity                     No            Yes   
4             Pre-seed       Equity                     No            Yes   

  Intnl founder?  Launched? Accelerator?  Employees  
0            Yes     2017.0           No       71.0  
1             No     2019.0           No       83.0  
2             No     2017.0          Yes       96.0  
3             No     2017.0           No        7.0  
4             No     2020.0           No        9.0

Based on the initial inspection of the dataset, we need to drop the  **'Unnamed: 1'** column.

In [6]:
df = df.drop(['Unnamed: 1'], axis=1) #removing the unnamed column from above, it has empty values
df.head(5)

Startup Location         Sector                Date  Amount (US$)  \
0  Daystar Power  Nigeria         energy      January + July    62000000.0   
1        uLesson  Nigeria        ed-tech  January + December    22500000.0   
2      Cowrywise  Nigeria        fintech             January     3000000.0   
3         Ogelle  Nigeria  entertainment             January           NaN   
4         Contro  Nigeria       e-health             January       42000.0   

                                           Investors                Stage  \
0  Investment Fund for Developing Countries (IFU)...             Series B   
1  Owl Ventures, TLcom Capital, Founder Collectiv...  Series A / Series B   
2  Quona Capital, Sahil Lavingia, Tsadik Foundati...         Pre-Series A   
3                            African Discovery Group          Undisclosed   
4                                                NaN             Pre-seed   

   Debt/Equity Female co-founder/CEO? Local founder? Intnl founder?  \
0  Equity/debt                     No            Yes            Yes   
1       Equity                     No            Yes             No   
2       Equity                     No            Yes             No   
3       Equity                     No            Yes             No   
4       Equity                     No            Yes             No   

   Launched? Accelerator?  Employees  
0     2017.0           No       71.0  
1     2019.0           No       83.0  
2     2017.0          Yes       96.0  
3     2017.0           No        7.0  
4     2020.0           No        9.0

# EDA

In [7]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1173 entries, 0 to 1172
Data columns (total 14 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Startup                 564 non-null    object 
 1   Location                564 non-null    object 
 2   Sector                  564 non-null    object 
 3   Date                    457 non-null    object 
 4   Amount (US$)            310 non-null    float64
 5   Investors               545 non-null    object 
 6   Stage                   564 non-null    object 
 7   Debt/Equity             564 non-null    object 
 8   Female co-founder/CEO?  564 non-null    object 
 9   Local founder?          564 non-null    object 
 10  Intnl founder?          564 non-null    object 
 11  Launched?               564 non-null    float64
 12  Accelerator?            564 non-null    object 
 13  Employees               565 non-null    float64
dtypes: float64(3), object(11)
memory usage: 

Lets check the tail of this dataset

In [9]:
df.tail()

Startup Location Sector Date  Amount (US$) Investors Stage Debt/Equity  \
1168     NaN      NaN    NaN  NaN           NaN       NaN   NaN         NaN   
1169     NaN      NaN    NaN  NaN           NaN       NaN   NaN         NaN   
1170     NaN      NaN    NaN  NaN           NaN       NaN   NaN         NaN   
1171     NaN      NaN    NaN  NaN           NaN       NaN   NaN         NaN   
1172     NaN      NaN    NaN  NaN           NaN       NaN   NaN         NaN   

     Female co-founder/CEO? Local founder? Intnl founder?  Launched?  \
1168                    NaN            NaN            NaN        NaN   
1169                    NaN            NaN            NaN        NaN   
1170                    NaN            NaN            NaN        NaN   
1171                    NaN            NaN            NaN        NaN   
1172                    NaN            NaN            NaN        NaN   

     Accelerator?  Employees  
1168          NaN        NaN  
1169          NaN        NaN  
1170          NaN        NaN  
1171          NaN        NaN  
1172          NaN        0.0

We have around 564 Non-null values for 1173 rows. We can assume that after row 564, we have NULL value. We'll have to investigate this. 

In [11]:
df.iloc[[560,561,562,563,564,565,566,567]]

Startup  Location       Sector     Date  Amount (US$)  \
560  OKO Finance      Mali      fintech    April     1200000.0   
561       alsoug     Sudan   e-commerce  October     5000000.0   
562       Ramani  Tanzania  retail-tech      NaN           NaN   
563        Gozem      Togo    transport  October     5000000.0   
564          NaN       NaN          NaN      NaN           NaN   
565          NaN       NaN          NaN      NaN           NaN   
566          NaN       NaN          NaN      NaN           NaN   
567          NaN       NaN          NaN      NaN           NaN   

                                             Investors        Stage  \
560  Newfund, ResiliAnce, Mercy Corps Ventures, Tec...         Seed   
561  Fawry, Rally Cap Ventures, Musha Ventures, others     Series A   
562             Launch Africa Ventures, Musha Ventures  Undisclosed   
563  AAIC, Thunes (TransferTo), Momentum Ventures (...     Series A   
564                                                NaN          NaN   
565                                                NaN          NaN   
566                                                NaN          NaN   
567                                                NaN          NaN   

    Debt/Equity Female co-founder/CEO? Local founder? Intnl founder?  \
560      Equity                     No             No            Yes   
561      Equity                    Yes            Yes             No   
562      Equity                     No            Yes             No   
563      Equity                     No            Yes             No   
564         NaN                    NaN            NaN            NaN   
565         NaN                    NaN            NaN            NaN   
566         NaN                    NaN            NaN            NaN   
567         NaN                    NaN            NaN            NaN   

     Launched? Accelerator?  Employees  
560     2017.0          Yes       12.0  
561     2016.0           No       11.0  
562     2019.0          Yes       55.0  
563     2018.0           No      114.0  
564        NaN          NaN        NaN  
565        NaN          NaN        NaN  
566        NaN          NaN        NaN  
567        NaN          NaN        NaN

The above dataset proves our earlier suspicion that our data has **564** rows instead of the **1173** rows.

**We need to select a subset of ROWS from 0-563** before we continue..

### subset with 564 rows

In [15]:
subset_df = df.iloc[0:564] #the range is not inclusive of the 564th row.
subset_df.tail()

Startup  Location       Sector     Date  Amount (US$)  \
559  AlphaDirect  Botswana      fintech    April      600000.0   
560  OKO Finance      Mali      fintech    April     1200000.0   
561       alsoug     Sudan   e-commerce  October     5000000.0   
562       Ramani  Tanzania  retail-tech      NaN           NaN   
563        Gozem      Togo    transport  October     5000000.0   

                                             Investors         Stage  \
559  Launch Africa Ventures, Century Oak Capital, a...  Pre-Series A   
560  Newfund, ResiliAnce, Mercy Corps Ventures, Tec...          Seed   
561  Fawry, Rally Cap Ventures, Musha Ventures, others      Series A   
562             Launch Africa Ventures, Musha Ventures   Undisclosed   
563  AAIC, Thunes (TransferTo), Momentum Ventures (...      Series A   

    Debt/Equity Female co-founder/CEO? Local founder? Intnl founder?  \
559      Equity                     No             No            Yes   
560      Equity                     No             No            Yes   
561      Equity                    Yes            Yes             No   
562      Equity                     No            Yes             No   
563      Equity                     No            Yes             No   

     Launched? Accelerator?  Employees  
559     2015.0           No        4.0  
560     2017.0          Yes       12.0  
561     2016.0           No       11.0  
562     2019.0          Yes       55.0  
563     2018.0           No      114.0

## TASK 1:What is the right time to seek funding?


## TASK 2: Do women-led startups get much less funding than men?

## TASK 3:Which sector is funded the most?

## TASK 4:What is the average age of these startups?

## TASK 5:Do accelerators increase the chances of funding and the amount?

## TASK 6:Which investors spends the most amount of money in Africa?
#### this can be broken down into country and  regions